c10/xpu/XPUFunctions.cpp

```diff
@@ -0,0 +1,188 @@
+#include <c10/util/CallOnce.h>
+#include <c10/util/Exception.h>
+#include <c10/xpu/XPUFunctions.h>
+
+#include <sys/wait.h>
+#include <unistd.h>
+#include <cmath>
+#include <deque>
+#include <mutex>
+#include <vector>
+
+namespace c10::xpu {
+namespace {
+
+/*
+ * Note [Device Management]
+ *
+ * An Intel GPU device qualifies as a type of SYCL device. This classification
+ * allows for the runtime querying of Intel GPU device information through the
+ * SYCL runtime library.
+ *
+ * Device status is managed through a SYCL device pool, with SYCL devices
+ * determined at runtime. There's currently a SYCL device pool that is lazily
+ * created and only initialized once, ensuring thread-local safety. Each device
+ * within the device pool shares the same default context.
+ */
+static c10::once_flag init_flag;
+static thread_local DeviceIndex curDeviceIndex = 0;
+
+struct DevicePool {
+  std::vector<std::unique_ptr<sycl::device>> devices;
+  std::unique_ptr<sycl::context> context;
+} gDevicePool;
+
+static void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
```

Comment:
You are already in anonymous namespace, so everything is already static, isn't it?
```suggestion
void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
```

Reply:
done.